File: Homework/Ngan_Luu/191-number-of-1-bits.py
#https://leetcode.com/problems/number-of-1-bits/
#Time: 	O(log2(n)-1)
#Space: O(1)
class Solution:
    def hammingWeight(self, n: int) -> int:
        count=0
        while(n):
            count+=(n&1)
            n>>=1
        return count

File: Homework/Son/W1/191-number-of-1-bits.js
// https://leetcode.com/problems/number-of-1-bits/submissions/

/**
 * @param {number} n - a positive integer
 * @return {number}
 */
 var hammingWeight = function(n) {
    // totalCount = 0;
    // let text = n.toString(2)
    // for (let i = 0; i< text.length; i++) {
    //     if (text[i] == 1) totalCount++;
    // }
    // return totalCount;
    
    let sum  =0;
    while (n != 0) {
        sum++
        n &= (n-1);
    }
    return sum;
};